admin/services/data-access/school-summary.data.service.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const { TYPES } = require('./sql.service')
+const sqlService = require('./sql.service')
+const R = require('ramda')
+
+const service = {}
+
+/**
+ * @description fetches register summary data from SQL data store
+ * @param {number} schoolId
+ * @return {Promise<object>}
+ */
+service.getRegisterData = async (schoolId) => {
+  const schoolIdParam = {
+    name: 'schoolId',
+    type: TYPES.Int,
+    value: schoolId
+  }
+  const sql = `
+    SELECT
+      COUNT(p.id) as [TotalCount],
+      SUM(CASE WHEN p.checkComplete = 1 AND p.attendanceId IS NULL THEN 1 ELSE 0 END) as [Completed],
+      SUM(CASE WHEN p.attendanceId IS NOT NULL THEN 1 ELSE 0 END) as [NotAttending],
+      MIN(s.name) as [schoolName],
+      MIN(s.dfeNumber) as [dfeNumber]
+    FROM
+      [mtc_admin].[pupil] p
+      INNER JOIN [mtc_admin].school s ON p.school_id = s.id
+    WHERE p.school_id = @schoolId`
+  const result = await sqlService.readonlyQuery(sql, [schoolIdParam])
+  return R.head(result)
+}
+
+service.getLiveCheckData = async (schoolId) => {
+  const schoolIdParam = {
+    name: 'schoolId',
+    type: TYPES.Int,
+    value: schoolId
+  }
+  const sql = `
+    SELECT
+        MIN(convert(varchar, c.createdAt, 106)) as [Date],
+        COUNT(c.id) AS [PinsGenerated],
+        SUM(CASE cs.code WHEN 'COL' THEN 1 ELSE 0 END) as [LoggedIn],
```

Comment:
I'm not sure about this as it is counting the 'current' state.   I would count all complete checks in the LoggedIn figure.  It must be the case!  

I propose:

```sql
SELECT
    MIN(convert(varchar, c.createdAt, 106)) as [Date],
    COUNT(c.id) AS [PinsGenerated],
    SUM(CASE WHEN c.pupilLoginDate IS NOT NULL THEN 1 ELSE 0 END) as [LoggedIn],
    SUM(CAST(c.complete AS INT)) as [Complete]
    FROM
        [mtc_admin].[check] c
        INNER JOIN [mtc_admin].pupil p ON (p.currentCheckId = c.id)
        LEFT JOIN [mtc_admin].[checkStatus] cs ON (c.checkStatus_id = cs.id)
    WHERE p.school_id = @schoolId AND c.isLiveCheck = 1
    GROUP BY cast(c.createdAt as date);
```

Reply:
thanks for the suggestion 👌